src/lib.rs

```diff
@@ -52,6 +52,32 @@ use traits::{
   AbsorbInROTrait, Engine, ROConstants, ROConstantsCircuit, ROTrait,
 };
 
+/// A type that holds parameters for the primary and secondary circuits of Nova and SuperNova
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct CircuitShape<E: Engine> {
```

Comment:
Nit: `CircuitShape` is a bit ambiguous with `R1CSShape` (since R1CS is also a circuit). Is this really required? If it is really needed, should we use a more descriptive name (e.g., `R1CSShapeWithArity`)?